### PR TITLE
Refactor export part 1 - export parameters

### PR DESCRIPTION
**Pull request by vinnamkim**
### Summary
- Ticket no. 137997
- We've been maintaining both `OTXModel._exporter` and `OTXModel._export_parameters` properties for the model exportation.
- However, this structure makes a model developer hard to know what to do to make their model to be exportable. For example, `OTXTVModel` develop should have fill in this flexible data object, `OTXModel._export_parameters` without any help of intellisense.
https://github.com/openvinotoolkit/training_extensions/blob/ce4b7fcb348c9f2b43c478dc0559b59d2e985a09/src/otx/algo/classification/torchvision_model.py#L285-L306
- This PR aims to model developers only care about the `OTXModel._exporter` property when they develop a new model. This will help them to fill in the appropriate fields with help of intellisense such as
image

- It can be done by revisiting `OTXModel._export_parameters` and extracting a strict data type object, `TaskLevelExportParameters` from a subset of them.
- `TaskLevelExportParameters` tries to package what can really be shared by task. Therefore, individual model developers don't need to care about it and can just pass it to `OTXModelExporter.__init__(task_level_export_parameters=...)`.
- Other export parameters can be filled in the derived property of `OTX_Model._exporter` of each individual model.

### How to test
Fixed some tests according to the source changes as well.

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: src/otx/algo/classification/mobilenet_v3_large.py [hunk 1]
```diff
@@ -48,11 +51,21 @@ def __init__(
...
+        mean, std = get_mean_std_from_data_processing(self.config)
+        return OTXNativeModelExporter(
+            task_level_export_parameters=self._export_parameters,
+            input_size=self.image_size,
+            mean=mean,
+            std=std,
+            resize_mode="standard",
+            pad_value=0,
+            swap_rgb=False,
+            via_onnx=True,  # NOTE: This should be done via onnx
```
Comment (sovrasov) [inline src/otx/algo/classification/mobilenet_v3_large.py]: This comment makes no sense now, since it doesn't reveal why do we need to use onnx here.
I'd change this to todo: check if this model can be exported directly with OV > 2024.0

File: src/otx/algo/classification/mobilenet_v3_large.py [hunk 2]
```diff
@@ -83,11 +96,21 @@ def __init__(
...
+            onnx_export_configuration=None,
+            output_names=["logits", "feature_vector", "saliency_map"] if self.explain_mode else None,
```
Comment (GalyaZalesskaya) [inline src/otx/algo/classification/mobilenet_v3_large.py:112]: For `OTXNativeModelExporter` here and further it should be:
`output_names=["logits", "feature_vector", "saliency_map"] if self.explain_mode else ["logits"],`

**Review by GalyaZalesskaya — COMMENTED** [review]
Thank you for the update

File: src/otx/algo/classification/mobilenet_v3_large.py (hunk 2 above)
Comment (sovrasov) [reply src/otx/algo/classification/mobilenet_v3_large.py:112]: None also works, but for classification `logits` is preferable

Comment (vinnamkim) [reply src/otx/algo/classification/mobilenet_v3_large.py:112]: Fixed in https://github.com/openvinotoolkit/training_extensions/pull/3289/commits/63f5c60423bcf2abc90f9ec0807a593c325a497a

File: src/otx/algo/classification/mobilenet_v3_large.py (hunk 1 above)
Comment (vinnamkim) [reply src/otx/algo/classification/mobilenet_v3_large.py]: Fixed in https://github.com/openvinotoolkit/training_extensions/pull/3289/commits/63f5c60423bcf2abc90f9ec0807a593c325a497a

**Review by jaegukhyun — DISMISSED** [review]

File: src/otx/algo/classification/mobilenet_v3_large.py (hunk 2 above)
Comment (vinnamkim) [reply src/otx/algo/classification/mobilenet_v3_large.py:112]: Revert 
```
output_names=["logits", "feature_vector", "saliency_map"] if self.explain_mode else None
=>
output_names=["logits", "feature_vector", "saliency_map"] if self.explain_mode else ["logits"]
```
because many errors happened
https://github.com/openvinotoolkit/training_extensions/actions/runs/8641394317/job/23690872757#step:5:4447

**Review by wonjuleee — APPROVED** [review]
LGTM

**Review by sovrasov — APPROVED** [review]
